pyoptsparse/pySNOPT/pySNOPT.py

```diff
@@ -612,6 +612,20 @@ def _set_snopt_options(self, iPrint, iSumm, cw, iw, rw):
 
         return
 
+    def _on_setOption(self, name, value):
```

Comment:
Yeah as I commented before I suppose these need to be here as placeholders to not trigger the error in the parent class right? Is this really best practice?

Reply:
This is not necessarily the best practice, but is done more common in other more traditional OOP languages such as C/C++, and is what's usually referred to as an "abstract class" where you define the functions that should be implemented, interfaces, class fields etc without actually implementing anything, leaving all to the inherited classes. This is not exactly Pythonic, we can change it so that the default implementation in Optimizer would simply "pass" everything, and any class that needs more can implement their own. That way, we do not need each inherited class to define the function unless they need something extra, and the base implementation would not throw errors. I will do this.

Reply:
I confirm that  I am onboard with [this commit](https://github.com/mdolab/pyoptsparse/pull/193/commits/868f71259eac4103153b358c73bb52eb9689d5d3), noice!